_drafts/A BLE Primer.md

```diff
@@ -0,0 +1,1202 @@
+
+Wireless technologies are a dime a dozen, each with their own quirks and features. Some have been around for a long time, and some have only popped up recently. We'll spare the comparison between the different ones to a future post, but some of the most interesting and applicable wireless technologies to embedded developers are:
+
+- Wi-Fi
+- Bluetooth Classic & Bluetooth Low Energy
+- Zigbee
+- Z-Wave
+- LoRaWAN
+- ANT/ANT+
+- NFC
+- Infrared (IR)
+
+Most of these focus on the low-power aspect and are suitable for embedding into devices that need to run on small batteries for long periods of time. Applications include: industrial sensors, medical devices, wearables, and more.
+
+In today's post, we'll be covering one of the most popular low-power IoT technologies: Bluetooth Low Energy (BLE). We'll go over:
+
+- The basics of Bluetooth Low Energy
+- A comparison between BLE and Bluetooth Classic
+- Benefits and Drawbacks
+- BLE Peripherals and Centrals
+- Advertising
+- Connections
+- The Generic Access Profile (GAP)
+- The Generic Attribute Profile (GATT), including Services and Characteristics
+- Implementation of a simple BLE Peripheral application on the nRF52 development kit
+- Interacting with the BLE Peripheral from a mobile phone
+
+## Basics of Bluetooth Low Energy (BLE)
+Bluetooth Low Energy (BLE) is a low power wireless technology used for connecting devices with each other. BLE operates in the **2.4 GHz ISM (Industrial, Scientific, and Medical)** band, and is targeted towards applications that need to consume less power and may need to run on batteries for longer periods of time — months, and even years.
+
+Bluetooth started as a short-distance cable replacement technology. For example, to replace wires in devices such as a mouse, keyboard, or a PC communicating with a Personal Digital Assistant (PDA) which were popular in the late 1990s and early 2000s. The first official version of Bluetooth was released by Ericsson in 1994, named after King Harald “Bluetooth” Gormsson of Denmark who helped unify warring factions in the 10th century CE.
+
+Bluetooth Low Energy (BLE), however, was introduced in the 4.0 version of the Bluetooth specification in 2010. The original Bluetooth defined in the previous versions is referred to as Bluetooth Classic. BLE was not an upgrade to the original Bluetooth: Bluetooth Classic, but rather it’s a new technology that utilizes the Bluetooth brand but focuses on the Internet of Things (IoT) applications where small amounts of data are transferred at lower speeds. It’s important to note that there’s a big difference between Bluetooth Classic and Bluetooth Low Energy in terms of technical specification, implementation and the types of applications they’re each suitable for.
```

Comment:
"BLE was not an upgrade to ~~the original Bluetooth~~ Bluetooth Classic, but rather..."

Reply:
Will fix.